### ISSUE TITLE
Concurrent map writes in fallthrough.go

### ISSUE THREAD

**Issue by divan** [edgedb/edgedb-go#200]
I've just upgraded to 0.9.2 and started to see random panics witth the following stacktrace:

```
fatal error: concurrent map writes

goroutine 135 [running]:
runtime.throw({0x10503c557, 0x15})
	/usr/local/go/src/runtime/panic.go:1198 +0x54 fp=0x14000646e50 sp=0x14000646e20 pc=0x104b323d4
runtime.mapassign_faststr(0x105201400, 0x14000342690, {0x14000638660, 0x1a})
	/usr/local/go/src/runtime/map_faststr.go:294 +0x3d0 fp=0x14000646ec0 sp=0x14000646e50 pc=0x104b0fac0
github.com/edgedb/edgedb-go.(*protocolConnection).fallThrough(0x14000101680, 0x1400034b300)
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/fallthrough.go:39 +0x514 fp=0x14000646fd0 sp=0x14000646ec0 pc=0x104da6084
github.com/edgedb/edgedb-go.(*protocolConnection).authenticate(0x14000101680, 0x1400034b300, 0x140001c4f30)
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/connect.go:240 +0xc60 fp=0x14000647140 sp=0x14000646fd0 pc=0x104d9a810
github.com/edgedb/edgedb-go.(*protocolConnection).connect(0x14000101680, 0x1400034b300, 0x140001c4f30)
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/connect.go:106 +0xf20 fp=0x14000647290 sp=0x14000647140 pc=0x104d99a30
github.com/edgedb/edgedb-go.connectWithTimeout({0x1052cce60, 0x14000194810}, 0x140001c4f30, {0x14000342690, 0x140002fc7d0, 0x140002fc820, 0x140002fc870, 0x140002fc8c0})
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/edgedb.go:79 +0x310 fp=0x14000647330 sp=0x14000647290 pc=0x104da3a80
github.com/edgedb/edgedb-go.(*reconnectingConn).reconnect(0x140004d80a0, {0x1052cce60, 0x14000194810}, 0x0)
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/reconnect.go:60 +0x240 fp=0x14000647430 sp=0x14000647330 pc=0x104db8af0
github.com/edgedb/edgedb-go.(*Client).newConn(0x14000253600, {0x1052cce60, 0x14000194810})
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/client.go:119 +0x13c fp=0x140006474e0 sp=0x14000647430 pc=0x104d96e8c
github.com/edgedb/edgedb-go.(*Client).acquire(0x14000253600, {0x1052cce60, 0x14000194810})
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/client.go:182 +0x3e0 fp=0x14000647820 sp=0x140006474e0 pc=0x104d972e0
github.com/edgedb/edgedb-go.(*Client).Query(0x14000253600, {0x1052cce60, 0x14000194810}, {0x1050563bb, 0x4e}, {0x1051bae40, 0x140004c6138}, {0x0, 0x0, 0x0})
	/Users/divan/pkg/mod/github.com/edgedb/edgedb-go@v0.9.2/client.go:281 +0x70 fp=0x14000647930 sp=0x14000647820 pc=0x104d98220
github.com/sportity/8lanes/server/db.(*DB).AllAges(0x140001a68e8, {0x1052cce60, 0x14000194810})
	/Users/divan/src/github.com/sportity/8lanes/server/db/age.go:60 +0x84 fp=0x140006479c0 sp=0x14000647930 pc=0x104dc8024
main.(*AgesService).List(0x14000114070, {0x1052cce60, 0x14000194810}, 0x14000194870)
	/Users/divan/src/github.com/sportity/8lanes/server/grpc_age.go:40 +0x3c fp=0x14000647a90 sp=0x140006479c0 pc=0x105022bcc
github.com/sportity/8lanes/server/pb._AgeService_List_Handler.func1({0x1052cce60, 0x14000194810}, {0x10523d500, 0x14000194870})
	/Users/divan/src/github.com/sportity/8lanes/server/pb/ages_grpc.pb.go:155 +0x7c fp=0x14000647ae0 sp=0x14000647a90 pc=0x104fbbd0c
main.(*AuthInterceptor).Unary.func1({0x1052cce60, 0x14000194810}, {0x10523d500, 0x14000194870}, 0x140004de000, 0x140004c6120)
	/Users/divan/src/github.com/sportity/8lanes/server/grpc_auth.go:93 +0xc4 fp=0x14000647b40 sp=0x14000647ae0 pc=0x105024594
github.com/sportity/8lanes/server/pb._AgeService_List_Handler({0x10524d580, 0x14000114070}, {0x1052cce60, 0x14000194810}, 0x140003e4120, 0x1400012c2b0)
	/Users/divan/src/github.com/sportity/8lanes/server/pb/ages_grpc.pb.go:157 +0x150 fp=0x14000647bb0 sp=0x14000647b40 pc=0x104fbbbc0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000162000, {0x1052d6580, 0x14000160a20}, 0x1400015c400, 0x1400012e930, 0x1056c4878, 0x0)
	/Users/divan/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1217 +0xc38 fp=0x14000647e30 sp=0x14000647bb0 pc=0x104fa8368
google.golang.org/grpc.(*Server).handleStream(0x14000162000, {0x1052d6580, 0x14000160a20}, 0x1400015c400, 0x0)
	/Users/divan/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1540 +0xa34 fp=0x14000647f60 sp=0x14000647e30 pc=0x104fac264
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400011e540, 0x14000162000, {0x1052d6580, 0x14000160a20}, 0x1400015c400)
	/Users/divan/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:878 +0x94 fp=0x14000647fb0 sp=0x14000647f60 pc=0x104fa6254
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1133 +0x4 fp=0x14000647fb0 sp=0x14000647fb0 pc=0x104b64b14
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/divan/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:876 +0x1f0
```

Looks like the culprit is this line: https://github.com/edgedb/edgedb-go/blob/52efdb658c8797750a398d634499bee829a7c07e/fallthrough.go#L39

I don't know what triggers the crash exactly (I just use the same code as used all the time), looks like it's just a nominal reconnect.

Does it provide enough information, or should I add more details / reprodicible code / full stacktrace?

Go version: 1.17.3
EdgeDB version: 1.0+9236fa8

**Comment by fmoor** [comment]
Thanks @divan! If you have a sample repo that reproduces this it would be great, but I'll do my best with this.

**Comment by rdlaitila** [comment]
I've hit this issue as well when conducting synthetic load testing on our app. Can trigger the condition with as little as 20 concurrent users via `siege`

```
siege -c20 -t30s <url>
```

**Comment by fmoor** [comment]
published [v0.9.3](https://github.com/edgedb/edgedb-go/releases/tag/v0.9.3) with a fix for this
